File: LongestPalindromicSubstring/LongestPalindromicSubstring.js
/**
 * @param {string} s
 * @return {string}
 */
var longestPalindrome = function(s) {
  let arr = [...s];
  let right = arr.length - 1;
  let longestPalindrome = "";

  for (let i = 0; i < arr.length; i++) {
    right = arr.length - 1;
    while (right >= i) {
      if (checkForPalindrome(arr, i, right) == true) {
        if (longestPalindrome.length < (right - i) + 1) {
          longestPalindrome = "";
          for (let j = i; j <= right; j++) {
            longestPalindrome += arr[j];
          }
        }
        break;
      }
      right--
    }
  }
  return longestPalindrome;

};

function checkForPalindrome(arr, left, right) {
  let isPalindrome = true;
  while (left <= right) {
    if (arr[left] != arr[right]) {
      isPalindrome = false;
      break;
    }
    left++;
    right--;
  }
  return isPalindrome;
}

s = [
  "babad",
  "cbbd",
  "xpsometemosldabalearrozalazorraelabady"
];


s.forEach(element => {
  console.log(longestPalindrome(element));
});

File: LongestPalindromicSubstring/longestPalindromicSubstring.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        longestLength = 0
        palindrome = ""

        left2 = 0
        right2 = 0

        for index in range(len(s)):
            left, right = self.isPalindrome(s, index, index)
            if (right - left + 1 > longestLength):
                longestLength = right - left + 1
                palindrome = s[left:right + 1]
            
            left2, right2 = self.isPalindrome(s, index, index + 1)
            if (right2 - left2 + 1 > longestLength):
                longestLength = right2 - left2 + 1
                palindrome = s[left2:right2 + 1]

        return palindrome
    
    def isPalindrome(self, s, left, right):
        while left >= 0 and right <= len(s) - 1 and s[left] == s[right]:
            left -= 1
            right += 1
        return left + 1, right - 1
    

s = ["babad", "cddf", "ptatk"]
sol = Solution()
for item in s:
  print(sol.longestPalindrome(item))